backend/src/api/classes/dto/existing-class-extended.dto.ts

```diff
@@ -0,0 +1,30 @@
+import { ApiProperty } from "@nestjs/swagger";
+import { Exclude, Expose, Transform } from "class-transformer";
+import { ExistingClassTeacherDto } from "./existing-class-teacher.dto";
+
+type SessionList = { id: number }[];
+
+export class ExistingClassExtendedDto extends ExistingClassTeacherDto {
+  @ApiProperty({
+    description: "The list of session IDs.",
+    type: [Number],
+    example: [1, 2],
+  })
+  @Transform(
+    ({ value }: { value: SessionList }) =>
+      value?.map((s: { id: number }) => s.id) ?? [],
+    { toClassOnly: true },
+  )
+  readonly sessions!: number[];
+
+  @ApiProperty({
+    name: "studentCount",
+    description: "The number of students in the class.",
+    type: Number,
+    example: 25,
+  })
+  @Expose({ name: "studentCount", toPlainOnly: true })
+  // Receive _count { students: number }, turn it into studentCount: number
+  @Transform(({ value }) => value?.students ?? 0, { toClassOnly: true })
+  readonly _count!: number;
```

Comment:
how could we handle situations with multiple counts?

Reply:
I'll cross that bridge (and revisit this) when I get there 😅
I'd originally done something different (decorator with `_count`, field named `studentCount`), but it didn't work the way I expected.
I'm sure it can be done, I just haven't gotten around to it.

Reply:
I've amended this to provide a cleaner solution, and it seems to be working too ;-)